### PR TITLE
Enhance HTTP Compliance CRLF modes

### PR DESCRIPTION
**Pull request by gregw**
Added modes to allow strict requirement for CRLF termination of headers and/or chunks

### CODE REVIEW

File: jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpCompliance.java [hunk 1]
```diff
@@ -166,11 +176,22 @@ public String getDescription()
...
     public static final String VIOLATIONS_ATTR = ComplianceViolation.CapturingListener.VIOLATIONS_ATTR_KEY;
 
+    /**
+     * The HttpCompliance mode that supports no known violations.
+     */
+    public static final HttpCompliance STRICT = new HttpCompliance("STRICT", noneOf(Violation.class));
+
+    /**
+     * The HttpCompliance mode that supports <a href="https://tools.ietf.org/html/rfc9110">RFC 9110</a>
+     * with no known violations.
```
Comment (sbordet) [inline jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpCompliance.java]: There is 1 violation.

File: jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpCompliance.java [hunk 2]
```diff
@@ -166,11 +176,22 @@ public String getDescription()
...
+     */
+    public static final HttpCompliance RFC9110 = new HttpCompliance("RFC9110", of(Violation.LF_HEADER_TERMINATION));
+
     /**
      * The HttpCompliance mode that supports <a href="https://tools.ietf.org/html/rfc7230">RFC 7230</a>
      * with no known violations.
```
Comment (sbordet) [inline jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpCompliance.java]: There are 2 violations.

**Review by sbordet — CHANGES_REQUESTED** [review]
1. What happens now if we have a request with mixed EOL, e.g. some are just \n, and some \r\n, still good and we have tests?
2. Please update the documentation about the compliance, defaulting to 9110 where now is 7230 in `programming-guide/server/compliance`.
3. This must be backported to 12.0.x, so I would rather rebase this PR on 12.0.x and then merge forward.

**Comment by gregw** [comment]
> 1. What happens now if we have a request with mixed EOL, e.g. some are just \n, and some \r\n, still good and we have tests?

If configured to accept LF EOL, then we accept a mix of LF and CRLF.    I guess we could have a semi strict mode that once you see a LF then any CRLF is an error... but that's not what the RFC says.   If you want to be strict, then just  accept CRLF only.

> 2. Please update the documentation about the compliance, defaulting to 9110 where now is 7230 in `programming-guide/server/compliance`.

Done.

> 3. This must be backported to 12.0.x, so I would rather rebase this PR on 12.0.x and then merge forward.

I'm not sure we need to backport.   There is no actual issue with jetty here and this is just tightening the compliance as a future proofing exercise.  Let's talk offline.

**Review by sbordet — APPROVED** [review]

**Pull request merged**